web/src/typing.ts

```diff
@@ -108,7 +151,24 @@ function notify_server_stop(to: Recipient): void {
     send_typing_notification_based_on_message_type(to, "stop");
 }
 
-export function get_recipient(): Recipient | null {
+function notify_server_editing_start(to: Recipient): void {
+    assert(to.notification_event_type === "typing_message_edit");
+    send_typing_notifications_for_message_edit(to.id, "start");
+}
+
+function notify_server_editing_stop(to: Recipient): void {
+    assert(to.notification_event_type === "typing_message_edit");
+    send_typing_notifications_for_message_edit(to.id, "stop");
+}
+
+export function get_recipient(message_id?: number): Recipient | null {
```

Comment:
`edting_message_id` might be a better name for this. But it might be better to just use a different function for the edit case.